src/main/java/appeng/parts/reporting/AbstractPartTerminal.java

```diff
@@ -69,6 +71,8 @@ public void getDrops(final List<ItemStack> drops, final boolean wrenched) {
                 drops.add(is);
             }
         }
+        ItemStack u = upgrades.getStackInSlot(0);
+        if (u != null) drops.add(u);
```

Comment:
This code is not called when a processing pattern terminal/pattern terminal is broken, so the upgrade doesn't drop.